File: 백준/Gold/12865. 평범한 배낭/평범한 배낭.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        int[][] items = new int[N][2];
        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            for (int j = 0; j < 2; j++) {
                items[i][j] = Integer.parseInt(st.nextToken());
            }
        }

        int[] dp = new int[K + 1];
        for (int i = 0; i < N; i++) {
            int weight = items[i][0];
            int value = items[i][1];

            for (int j = K; j >= weight; j--) {
                dp[j] = Math.max(dp[j], dp[j - weight] + value);
            }
        }

        System.out.println(dp[K]);
    }
}

File: 백준/Gold/12865. 평범한 배낭/평범한 배낭.py
N, K = map(int, input().split())
dp = [0 for _ in range(K + 1)]
items = []
for i in range(N):
    W, V = map(int, input().split())
    items.append((W, V))

for i in range(N):
    W, V = items[i]
    for j in range(K, W-1, -1):
        dp[j] = max(dp[j], dp[j - W] + V)

print(dp[K])